frontend/src/app/pages/exercises/exercise/shared/exercise-map/feature-managers/material-feature-manager.ts

```diff
@@ -61,10 +83,39 @@ export class MaterialFeatureManager extends MoveableFeatureManager<Material> {
             },
             new PointGeometryHelper()
         );
-        this.layer.setStyle((feature, resolution) => [
-            this.nameStyleHelper.getStyle(feature as Feature, resolution),
-            this.imageStyleHelper.getStyle(feature as Feature, resolution),
-        ]);
+        this.layer.setStyle((feature, resolution) => {
+            const styles = [
+                this.nameStyleHelper.getStyle(feature as Feature, resolution),
+                this.imageStyleHelper.getStyle(feature as Feature, resolution),
+            ];
+            if (
+                this.popupService.currentPopup?.markedForTrainerUUIDs.includes(
+                    feature.getId() as UUID
+                ) &&
+                selectStateSnapshot(selectCurrentRole, this.store) === 'trainer'
+            ) {
+                styles.push(
+                    this.openPopupCircleStyleHelper.getStyle(
+                        feature as Feature,
+                        resolution
+                    )
+                );
+            } else if (
+                this.popupService.currentPopup?.markedForParticipantUUIDs.includes(
+                    feature.getId() as UUID
+                ) &&
+                selectStateSnapshot(selectCurrentRole, this.store) ===
+                    'participant'
+            ) {
+                styles.push(
+                    this.openPopupCircleStyleHelper.getStyle(
+                        feature as Feature,
+                        resolution
+                    )
+                );
```

Comment:
The body of this `else` statement is the same as the one for the `if` statement. Please combine the two conditions.

(Should be automatically resolved when implementing the `super.markedForCurrentRole()` helper.)

Reply:
Done